pkgs/development/python-modules/dictdiffer/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, check-manifest
+, mock
+, pycodestyle
+, pydocstyle
+, pytest
+, pytest-cov
+, pytest-pycodestyle
+, pytest-pydocstyle
+, pytest-isort
+, pytest-runner
+, setuptools-scm
+, sphinx
+, tox
+}:
+
+buildPythonPackage rec {
+  pname = "dictdiffer";
+  version = "0.9.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-F7rPX7/mE8zxttUSvXZuayH7eYgioTOqhgmLismZdXg=";
+  };
+
+  buildInputs = [
+    check-manifest
+    mock
+    pycodestyle
+    pydocstyle
+    pytest
+    pytest-cov
+    pytest-pycodestyle
+    pytest-pydocstyle
+    pytest-isort
+    pytest-runner
+    setuptools-scm
+    sphinx
+    tox
+  ];
```

Comment:
Please split checkInputs. Packages needed at runtime need to go to propogatedBuildInputs.

pytest-cov,pytest-pycodestyle, pytest-pydocstyle are not required because they check code quality. Please patch them out.

setuptools-scm goes into nativeBuildInputs.

Reply:
also, use pytestCheckHook